src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -2192,10 +2192,24 @@ private static bool IsUserDefinedTrueOrFalse(BoundUnaryOperator @operator)
                         convertedNullability = convertedCollection.TopLevelNullability;
                         conversion = convertedCollectionConversion;
                     }
+                    else if (highestBoundExpr is BoundConversion { ConversionKind: ConversionKind.ImplicitNullable, Conversion.UnderlyingConversions: [{ Kind: ConversionKind.CollectionExpression }] } boundConversion)
+                    {
+                        convertedType = highestBoundExpr.Type;
+                        convertedNullability = convertedCollection.TopLevelNullability;
+                        conversion = boundConversion.Conversion;
+                    }
+                    else if (boundNodeForSyntacticParent is BoundConversion parentConversion)
+                    {
+                        // There was an explicit cast on top of the collection expression.
```

Comment:
In this case, I would not expect `ConvertedType` to contain something different than the current type. `ConvertedType` is explicitly documented as implicit conversions only.

Reply:
I didn't know that. Thanks for pointing it out!